apps/core/declarativewebcontainer.cpp

```diff
@@ -193,6 +194,16 @@ void DeclarativeWebContainer::setWebPage(DeclarativeWebPage *webPage, bool trigg
             connect(m_webPage.data(), &DeclarativeWebPage::domContentLoadedChanged,
                     this, &DeclarativeWebContainer::updateActiveTabRendered, Qt::UniqueConnection);
 
+            connect(m_webPage.data(), &DeclarativeWebPage::neterror, [this]() {
+                if (m_historyModel)
+                    m_historyModel->remove(m_webPage->url().toString());
+            });
+
+            connect(m_webPage.data(), &DeclarativeWebPage::urlChanged, [this]() {
+                if (!BrowserApp::captivePortal() && !m_privateMode && m_historyModel)
+                    m_historyModel->add(m_webPage->url().toString(), m_webPage->title());
+            });
```

Comment:
This same should be done for the title change. Sounds that update(url, title) would be good for that.

Reply:
Already handled by the historymodel itself.